src/StdCheats.sol

```diff
@@ -222,6 +233,84 @@ abstract contract StdCheatsSafe {
         assumeNotBlacklisted(token, addr);
     }
 
+    function assumeAddressIsNot(AddressType addressType, address addr) internal virtual {
+        if (addressType == AddressType.Payable) {
+            assumeNoPayable(addr);
+        } else if (addressType == AddressType.NonPayable) {
+            assumeNoNonPayable(addr);
+        } else if (addressType == AddressType.ZeroAddress) {
+            assumeNoZeroAddress(addr);
+        } else if (addressType == AddressType.Precompiles) {
+            assumeNoPrecompiles(addr);
+        } else if (addressType == AddressType.ForgeAddresses) {
+            assumeNoForgeAddresses(addr);
+        }
+    }
+
+    function assumeAddressIsNot(AddressType addressType1, AddressType addressType2, address addr) internal virtual {
+        assumeAddressIsNot(addressType1, addr);
+        assumeAddressIsNot(addressType2, addr);
+    }
+
+    function assumeAddressIsNot(
+        AddressType addressType1,
+        AddressType addressType2,
+        AddressType addressType3,
+        address addr
+    ) internal virtual {
+        assumeAddressIsNot(addressType1, addr);
+        assumeAddressIsNot(addressType2, addr);
+        assumeAddressIsNot(addressType3, addr);
+    }
+
+    function assumeAddressIsNot(
+        AddressType addressType1,
+        AddressType addressType2,
+        AddressType addressType3,
+        AddressType addressType4,
+        address addr
+    ) internal virtual {
+        assumeAddressIsNot(addressType1, addr);
+        assumeAddressIsNot(addressType2, addr);
+        assumeAddressIsNot(addressType3, addr);
+        assumeAddressIsNot(addressType4, addr);
+    }
+
+    function _isPayable(address addr) private returns (bool) {
+        uint256 size;
+        assembly {
+            size := extcodesize(addr)
+        }
+
+        if (size == 0) {
+            // return false if no code
+            return false;
+        } else {
+            require(addr.balance < UINT256_MAX, "balance exceeds max uint256");
+            uint256 origBalanceTest = address(this).balance;
+            uint256 origBalanceAddr = address(addr).balance;
+            (bool success,) = payable(addr).call{value: 1}("");
```

Comment:
This will fail if the test contract has zero ETH, and therefore may erroneously return `success = false` in that case, so we should `vm.deal(address(this), 1)` beforehand.

Also one side effect of this method is there may be state changes depending on the fallback/receive logic implemented by `addr`. We should make sure this is documented in comments here + in the book

Reply:
Yeah, that's a good point.